pkg/pub_integration/test/publisher_test.dart

```diff
@@ -43,8 +43,8 @@ void main() {
           },
           body: json.encode({'granted': true}),
         );
-        if (rs1.statusCode != 404) {
-          throw Exception('Expected status code 404, got: ${rs1.statusCode}');
+        if (rs1.statusCode != 400) {
```

Comment:
Why not
```
expect(rs1.statusCode, 400);
```
?

Reply:
The scripts are designed for dual nature: to run as test (against the fake server), and to run as a program (against the staging or production deployment). Although only a fraction is run against the real deployment, the pattern was to use throw-like checks instead of using `package:test`. We could probably add a helper method to make that simpler.

Reply:
Ok - I suspected there would be a reason.
But you still import package test?

Reply:
In this case it is part of a (test-specific) callback that gets evaluated as part of the script, and should throw an exception if the consent was not valid or could not be accepted. I think we could use a `package:test` here, but I'd rather keep it consistent with the script itself.